src/web/browser/ophan/ophan.ts

```diff
@@ -84,6 +84,12 @@ export type TestMeta = {
     products?: OphanProduct[];
 };
 
+export const submitComponentEvent = (
+    componentEvent: OphanComponentEvent,
+): void => {
+    window.guardian.ophan.record({ componentEvent });
+};
```

Comment:
I think you need to make sure that `window.guardian.ophan.record`, as is done on [line 137][L137].

```suggestion
): void => {
  if (
        window.guardian &&
        window.guardian.ophan &&
        window.guardian.ophan.record
    ) {
      window.guardian.ophan.record({ componentEvent });
    } else {
        throw new Error("window.guardian.ophan.record doesn't exist");
    }
};
```

[L137]: https://github.com/guardian/dotcom-rendering/pull/1753/files#diff-6bc07cd001d12de643e0551db421254eR137

Reply:
Thanks @mxdvl, we weren't making this check previously when calling`window.guardian.ophan.record` from `sendOphanComponentEvent` so didn't deem it necessary, however I appreciate the extra security of doing so.

Reply:
Looks like the third usage, so abstracting it out into a `record` function and using that might be beneficial!